src/components/TierPurchase/TierPurchase.js

```diff
@@ -12,142 +12,104 @@ import { func, instanceOf, node, number, string } from 'prop-types'
 import FarmhandContext from '../Farmhand/Farmhand.context'
 import './TierPurchase.sass'
 
-export class TierPurchase extends Component {
-  state = {
-    selectedTier: '',
-  }
+export function TierPurchase({
+  description,
+  handleTierPurchase,
+  maxedOutPlaceholder,
+  money,
+  purchasedTier,
+  renderTierLabel,
+  tiers,
+  title,
+}) {
+  const [selectedTier, setSelectedTier] = useState(purchasedTier)
 
-  tierValues = [...this.props.tiers.entries()]
+  const tierValues = [...tiers.entries()]
 
-  get selectedTierNumber() {
-    return Number(this.state.selectedTier)
-  }
+  const selectedTierNumber = Number(selectedTier)
+
+  const hasPurchasedTier = tierLevel => tierLevel <= purchasedTier
+  const hasPurchasedHighestTier = hasPurchasedTier(tierValues.slice(-1)[0][0])
+
+  if (hasPurchasedTier(selectedTier)) {
+    const nextTierNumber = selectedTierNumber + 1
+    const nextTierToPurchase = tiers.get(nextTierNumber)
 
-  get canPlayerBuySelectedTier() {
-    const {
-      props: { money },
-      selectedTierNumber,
-    } = this
+    if (nextTierToPurchase && money >= nextTierToPurchase.price) {
+      setSelectedTier(`${nextTierNumber}`)
+    }
+  }
 
-    const selectedTier = this.props.tiers.get(selectedTierNumber)
+  const canPlayerBuySelectedTier = () => {
+    const selectedTier = tiers.get(selectedTierNumber)
 
     return (
       !!selectedTier &&
-      !this.hasPurchasedTier(selectedTierNumber) &&
+      !hasPurchasedTier(selectedTierNumber) &&
       money >= selectedTier.price
     )
   }
 
-  hasPurchasedTier = tierLevel => tierLevel <= this.props.purchasedTier
+  const onClickBuy = () => {
```

Comment:
Would you be okay with changing this back to `handleTierPurchase`, or perhaps `handleBuyClick`? I've found that naming event handlers like `handleXEvent` helps to keep a distinction between prop and function names. 

This article does a pretty good job of defining a concrete style for this sort of thing: https://jaketrent.com/post/naming-event-handlers-react. I just found it, but if this seems reasonable to you I can add it to the documentation so we can start building up a style guide!

Reply:
> i actually only thought to rename this because `handleTierPurchase` is the prop and forced my hand to make a change.

Oh whoops... it looks like there's a number of preexisting issues with this file. 😅 I don't expect anyone else to clean up my mess, so don't stress about this. I think that to conform to the article, we'd need to expose an `onBuyClick` prop for this `TierPurchase` component. Changing this component's API would be blowing out the scope of this PR though, which I don't think is necessary.

Based on this part of the article:

> For the function names, I follow the exact same pattern, but I replace `on` with `handle*`, as in `handleClick`. Together, it’d look like:
>
> <MyComponent onClick={handleClick} />

I think this internal handler would need to be `handleBuyClick`. Does that seem right?

Reply:
none of this was used in _that_ many places so i went ahead and did the prop change. let me know though if i still don't have the method names correct!